### ISSUE TITLE
TextField misplaced placeholder (RTL)

### ISSUE THREAD

**Issue by Sufi-Al-Hussaini** [CosmicMind/Material#456]
Hey Daniel,

First off, thanks for this cool project!

I find that `TextField` placeholders are misplaced for RTL scripts. 
Is there a workaround I could use currently?

![TextField misplaced placeholder (RTL)](https://s31.postimg.org/kum831op7/Simulator_Screen_Shot_Jun_30_2016_4_39_28_PM.png)

**Comment by danieldahan** [comment]
Thank you :)

Can you share with me the setup code. If you set the `textField.textAlignment` it should align everything for you.

**Comment by Sufi-Al-Hussaini** [comment]
The screenshot above is from the `Storyboards -> TexField` example. Here's the code:

```
private func prepareEmailField() {
    emailField.placeholder = "Email"
    emailField.delegate = self
    emailField.detail = "Email is incorrect."
}
```

I tried setting its `textAlignment` to `.Natural` but it didn't work. It does work with `.Right` though. 
Currently, I'm subclassing `TextField` and using the following code, and it's working for me.

```
class MyTextField: TextField {
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        setCustomStyle()
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setCustomStyle()
    }

    convenience init() {
        self.init(frame: CGRectZero)
    }

    func setCustomStyle() {
        if (UIApplication.sharedApplication().userInterfaceLayoutDirection == .RightToLeft) {
            self.textAlignment = .Right
        }
        else {
            self.textAlignment = .Left
        }
}
```

Is this the right thing to do?

**Comment by danieldahan** [comment]
Yes, I would use the .Right or .Left, not .Natural. I am treating .Natural like .Left right now. I could update that, but for now that is what is happening. Subclassing is perfectly fine. Your code looks like it is doing the right thing. Another option though, which could save you some code, is to do this:

```swift
class MyTextField: TextField {
    override func prepareView() {
        super.prepareView()
        if (UIApplication.sharedApplication().userInterfaceLayoutDirection == .RightToLeft) {
            self.textAlignment = .Right
        }
        else {
            self.textAlignment = .Left
        }
}
```

**Comment by danieldahan** [comment]
Generally though, I think we could add the code you are adding internally to help with this issue and solve the problem for everyone.

**Comment by Sufi-Al-Hussaini** [comment]
Thanks a lot for your help @danieldahan .

**Comment by danieldahan** [comment]
Thank you for the PR :) 

I updated it to look like this:

```swift
/// Prepares the textAlignment.
    private func prepareTextAlignment() {
        textAlignment = UIApplication.sharedApplication().userInterfaceLayoutDirection == .RightToLeft ? .Right : .Left
    }
```

I will release as soon as I am home. Thanks again! I will close the issue then.

**Comment by Sufi-Al-Hussaini** [comment]
You're welcome. :)
Sure, please do...

**Comment by danieldahan** [comment]
Released :) [1.42.7](https://github.com/CosmicMind/Material/releases/tag/1.42.7). Thank you!
